### ISSUE TITLE
decimal lose precision digits to postgres

### ISSUE THREAD

**Issue by soooooot** [prisma/prisma#8160]
### Bug description

save `Prisma.Decimal("100000000000000000000000")` ,   result: `9999999999999999`
save `Prisma.Decimal("1234522345323454234552345")` , result: `234522345323454`

result length has been cut to 16.

However, the inserting with `raw sql` works. 
please check details in minimal reproduction repo.

postgres log:
```
LOG:  execute s3: INSERT INTO "public"."Price" ("id","price") VALUES ($1,$2), ($3,$4)
DETAIL:  parameters: $1 = '1', $2 = '9999999999999999', $3 = '2', $4 = '1234522345323454'  // <-- trimmed in generating sql
LOG:  statement: COMMIT
LOG:  execute s4: INSERT INTO "public"."Price" ("id", "price") VALUES (3, 100000000000000000000000), (4, 1234522345323454234552345) // <-- raw sql works
```

the issue is similar to #5925.

### How to reproduce

Reproduce with [minimal repoduction repo](https://github.com/soooooot/prisma-test/tree/prisma-decimals ), branch "prisma-decimals".

Run

```
yarn db:up
yarn
yarn prisma:generate
yarn migrate:deploy
yarn start
```

See the output.

### Expected behavior

save `Prisma.Decimal("100000000000000000000000")` ,   result: `100000000000000000000000`
save `Prisma.Decimal("1234522345323454234552345")` , result: `1234522345323454234552345`

### Prisma information

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Price {
  id        Int     @id
  price     Decimal @db.Decimal(100, 0)
}
```

### Environment & setup

- OS: MacOS
- Database: PostgreSQL 12.1 
- Node.js version: v14.17.0

### Prisma Version

```
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : darwin
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0

```

**Comment by reecefenwick** [comment]
Can also confirm I'm having this issue with MySQL, any updates on this?

**Comment by YaakovR** [comment]
Same issue for SQL Server. The value 97.85 turns into 97.8499999999999900.

**Comment by timpetri** [comment]
Does anyone have some kind of a workaround for this?

**Comment by reecefenwick** [comment]
@timpetri I included a workaround here https://github.com/prisma/prisma/issues/10412

I’ve stopped using prisma, there are too many issues like these that make it unsuitable for production IMO

**Comment by SevInf** [comment]
I know I've sent a bunch of people to this issue, saying that it fixes precision issues, not already fixed by  #15157. But turns out #15157 actually fixes all precision issues, so this issue can be closed as well.
